.vscode/tasks.json

```
@@ -157,6 +157,20 @@
         "kind": "build",
         "isDefault": true
       }
+    },
+    {
+      "label": "Install integration requirements",
+      "detail": "Install all requirements of a given integration.",
+      "type": "shell",
+      "command": "python3 -m script.install_integration_requirements ${input:integrationName}",
```

Tasks runs in a new shell, you need to make sure that you activate the venv before installing these if needed like we do in the setup script.
https://github.com/home-assistant/core/blob/dev/script/setup#L19

---

```suggestion
      "command": "script/run-in-env.sh python3 -m script.install_integration_requirements ${input:integrationName}",
```
It is untested, but that might do the trick.

---

I use `${command:python.interpreterPath} -m ...` for my personal tasks. That uses the environment activated in VSCode. `script/run-in-env.py` looks only for `.venv` environments and my not work with custom env names.